services/gossip/service.go

```diff
@@ -71,14 +89,48 @@ func (s *service) OnTransportMessageReceived(ctx context.Context, payloads [][]b
 	}
 
 	logger.Info("transport message received", log.Stringable("header", header), log.String("gossip-topic", header.StringTopic()))
-	switch header.Topic() {
-	case gossipmessages.HEADER_TOPIC_TRANSACTION_RELAY:
-		s.receivedTransactionRelayMessage(ctx, header, payloads[1:])
-	case gossipmessages.HEADER_TOPIC_LEAN_HELIX:
-		s.receivedLeanHelixMessage(ctx, header, payloads[1:])
-	case gossipmessages.HEADER_TOPIC_BENCHMARK_CONSENSUS:
-		s.receivedBenchmarkConsensusMessage(ctx, header, payloads[1:])
-	case gossipmessages.HEADER_TOPIC_BLOCK_SYNC:
-		s.receivedBlockSyncMessage(ctx, header, payloads[1:])
+	s.messageDispatcher.dispatch(logger, header, payloads[1:])
+}
+
+type gossipMessageDispatcher map[gossipmessages.HeaderTopic]chan gossipMessage
+
+// These channels are buffered because we don't want to assume that the topic consumers behave nicely
+// In fact, Block Sync should create a new one-off goroutine per "server request", Consensus should read messages immediately and store them in its own queue,
+// and Transaction Relay shouldn't block for long anyway.
+func makeMessageDispatcher() (d gossipMessageDispatcher) {
+	d = make(gossipMessageDispatcher)
+	d[gossipmessages.HEADER_TOPIC_TRANSACTION_RELAY] = make(chan gossipMessage, 10)
```

Comment:
I suggest making the buffer size configurable. We don't know if 10 is the optimal configuration.

Reply:
yagni, I think